internal/gossiper/target.go

```diff
@@ -323,47 +324,38 @@ func (g *Proposer[T]) Run(client *p2p.Client) {
 	}
 }
 
-func (g *Proposer[T]) BlockVerified(t int64) {
+func (g *Target[T]) BlockVerified(t int64) {
 	if t < g.lastVerified {
 		return
 	}
 	g.lastVerified = t
 }
 
-func (g *Proposer[T]) Done() {
+func (g *Target[T]) Done() {
 	g.timer.Stop()
 	<-g.doneGossip
 }
 
-func (g *Proposer[T]) sendTxs(ctx context.Context, txs []T) error {
+func (g *Target[T]) sendTxs(ctx context.Context, txs []T) error {
 	ctx, span := g.tracer.Start(ctx, "Gossiper.sendTxs")
 	defer span.End()
 
-	// Marshal gossip
-	b, err := g.serializer.Marshal(txs)
+	gossipContainers, err := g.targetStrategy.Target(ctx, txs)
 	if err != nil {
 		return err
 	}
 
-	// Select next set of proposers and send gossip to them
-	proposers, err := g.validatorSet.Proposers(
-		ctx,
-		g.cfg.GossipProposerDiff,
-		g.cfg.GossipProposerDepth,
-	)
-	if err != nil {
-		return fmt.Errorf("%w: unable to fetch proposers", err)
-	}
-	if proposers.Len() == 0 {
-		return errors.New("no proposers to gossip to")
-	}
-	recipients := set.NewSet[ids.NodeID](len(proposers))
-	for proposer := range proposers {
-		// Don't gossip to self
-		if proposer == g.validatorSet.NodeID() {
-			continue
+	for _, gossipContainer := range gossipContainers {
+		// Marshal gossip
+		b, err := g.serializer.Marshal(gossipContainer.Txs)
+		if err != nil {
+			return err
+		}
+
+		// Send gossip to specified peers
+		if err := g.client.AppGossip(ctx, gossipContainer.SendConfig, b); err != nil {
+			return err
 		}
```

Comment:
this section should be done in parallel, and with the error being accumulated and returned to the main goroutine.
i.e. an error related to one container shouldn't impact others.

Reply:
This is actually already using the p2p client to send a message via the AvalancheGo p2p networking library: https://github.com/ava-labs/avalanchego/blob/master/network/p2p/client.go#L116.

Errors from here are treated as fatal and this will only push the message to a queue, not block on sending the message, so I don't think there's a need to accumulate errors or parallelize here